Assets/Tests/BPMTest.cs

```diff
@@ -0,0 +1,65 @@
+﻿using NUnit.Framework;
+using System.Collections;
+using System.Linq;
+using Tests.Util;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Tests
+{
+    public class BPMTest
+    {
+        [UnityOneTimeSetUp]
+        public IEnumerator LoadMap() => TestUtils.LoadMapper();
+
+        [TearDown]
+        public void ContainerCleanup()
+        {
+            BeatmapActionContainer.RemoveAllActionsOfType<BeatmapAction>();
+            TestUtils.CleanupBPMChanges();
+        }
+
+        private static void CheckBPM(BeatmapObjectContainerCollection container, int idx, int time, int bpm)
+        {
+            var newObjA = container.LoadedObjects.Skip(idx).First();
+            Assert.IsInstanceOf<BeatmapBPMChange>(newObjA);
+            if (newObjA is BeatmapBPMChange newNoteA)
+            {
+                Assert.AreEqual(time, newNoteA._time);
+                Assert.AreEqual(bpm, newNoteA._BPM);
+            }
+        }
+        
+        [Test]
+        public void ModifyEvent()
+        {
+            var actionContainer = Object.FindObjectOfType<BeatmapActionContainer>();
+            var collection = BeatmapObjectContainerCollection.GetCollectionForType(BeatmapObject.Type.BPM_CHANGE);
+            if (collection is BPMChangesContainer bpmCollection)
+            {
+                var root = bpmCollection.transform.root;
```

Comment:
i am root

(is there a reason why we're getting the root here? i'm not seeing it used anywhere)

Reply:
Copy paste fail